### PR TITLE
Feature: Implementation of Omnibus Directive price history tracking

### LINKED ISSUE #11209: [2.5] Implementation of Omnibus Directive price history tracking
### Is your feature request related to a problem? Please describe.

Currently, Bagisto does not have a native mechanism to track and display historical prices. This makes it difficult for merchants operating in the EU to comply with the legal requirement of showing the lowest price from the last 30 days prior to a discount. Without this feature, merchants risk legal penalties and provide less transparency to their customers.

### Describe the solution you'd like

I have developed a core-ready implementation that includes:

Price Snapshot Engine: A combination of Eloquent Events and a scheduled Cron task to capture price points for all channels and currencies.

Configurable Product Handling: Full recursive support for tracking variant prices separately.

Omnibus Logic Helper: A specialized service that calculates the lowest price specifically for the 30-day window before the current promotion starts (avoiding logic loops).

Dynamic UI Integration: Vue 3 event listener (configurable-variant-selected-event) to update the Omnibus info instantly when a user selects a product variant.

Performance & Maintenance: Built-in garbage collector to prune records older than 35 days and optimized EAV queries for minimal database load.

Toggleable Configuration: I've added a new toggle switch in Configure -> Catalog -> Products (at the bottom of the settings). The feature is disabled by default to ensure no unexpected changes for existing stores.

Suggestion: It might be worth considering enabling this by default for installations localized in European regions to ensure immediate legal compliance.

Note: If I missed any specific core standards or if you'd like to see additional logic implemented, please let me know. I am ready to provide quick fixes or additional changes to get this merged into the core.

### Describe alternatives you've considered

Alternative 1: Using third-party extensions, but these often lack core integration, slow down the system with redundant queries, or fail to handle Bagisto's complex configurable product logic correctly.

Alternative 2: Manual price logging, which is not scalable for merchants with large catalogs and is prone to human error.

### Additional context

The solution is fully tested using Pest (Unit & Feature tests) according to the latest Bagisto standards.

It respects all active channel and currency settings.

Image

Image

Final Note:
I have done my best to cover all Omnibus requirements and Bagisto edge cases (like configurable products). However, if I missed any specific core standards or if you'd like to see additional logic implemented, please let me know. I am ready to provide quick fixes or additional changes to get this merged into the core.

### PR DESCRIPTION
**Pull request by SopsoN**
## Issue Reference
Closes #11209

## Description
This PR introduces a comprehensive price history tracking system to comply with the EU Omnibus Directive (2019/2161). 

**Key Changes:**
- **Price Tracking Engine:** Captures price changes via Eloquent events and a daily scheduled task (`omnibus:snapshot-prices`).
- **Configurable Product Support:** Implements recursive price logging for all product variants.
- **Frontend Integration:** Added a dynamic UI component that displays the lowest price from the last 30 days prior to the current discount.
- **Dynamic Updates:** Uses Vue 3 emitter to instantly update the Omnibus price information when a variant is selected on the product page.
- **Maintenance:** Built-in garbage collector to prune records older than 35 days and optimized database queries.
- **Configuration:** Added a toggle switch in `Configure -> Catalog -> Products` (disabled by default).

Screenshot 2026-03-23 at 17 40 00
Screenshot 2026-03-23 at 17 39 45

## How To Test This?
1. Enable the Omnibus feature in `Admin -> Configuration -> Catalog -> Products`.
2. Run the snapshot command: `php artisan omnibus:snapshot-prices`.
3. Create a special price for a product in the admin panel.
4. Visit the product page on the frontend; you should see the "Lowest price in the last 30 days" message.
5. For configurable products, change variants and verify that the Omnibus price updates dynamically.
6. Run the test suite: `php artisan test packages/Webkul/Omnibus` (if integrated into core tests).

## Documentation
- [ ] My pull request requires an update on the documentation repository.
## Branch Selection
- [x] Target Branch: master

## Tailwind Reordering
- [x] All Tailwind classes are reordered.

### CODE REVIEW

**Comment by devansh-webkul** [comment]
Allow me some time, will start on this soon.

File: composer.json [hunk 1]
```diff
@@ -104,7 +104,8 @@
             "Webkul\\Stripe\\": "packages/Webkul/Stripe/src",
             "Webkul\\Tax\\": "packages/Webkul/Tax/src",
             "Webkul\\Theme\\": "packages/Webkul/Theme/src",
-            "Webkul\\User\\": "packages/Webkul/User/src"
+            "Webkul\\User\\": "packages/Webkul/User/src",
+            "Webkul\\Omnibus\\": "packages/Webkul/Omnibus/src"
```
Comment (devansh-webkul) [inline composer.json]: Please add OmniBus according to alphabatical order.

File: composer.json
Comment (devansh-webkul) [inline composer.json:122]: Also omnibus autoload missing for testcases.

File: packages/Webkul/Omnibus/src/Resources/lang/uk/app.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    'admin' => [
+        'system' => [
+            'omnibus' => 'Omnibus',
+            'omnibus-info' => 'Omnibus directive configuration',
+            'is-enabled' => 'Enable Omnibus',
+        ],
+    ],
+
+    'shop' => [
+        'price-info' => 'Lowest price 30 days prior to the discount:',
```
Comment (devansh-webkul) [inline packages/Webkul/Omnibus/src/Resources/lang/uk/app.php]: Please update the exact translations for each file and now we have 22 languages in core.

**Comment by devansh-webkul** [comment]
Also please update the master branch with this. In master branch we now have two driver support i.e. MySQL and Postgres. Please check with both drivers werther your queries are working fine in both or not.

**Comment by devansh-webkul** [comment]
Then i will take a pull and provide you the final review for this.

**Comment by devansh-webkul** [comment]
Hello @SopsoN,

Any update on this?

**Comment by SopsoN** [comment]
Hello @devansh-webkul! 

Apologies for the delay. I've updated the branch with the latest master.

I hope all the changes are now implemented correctly. Could you please take a look, verify, and proceed with the final code review? Let me know if anything else is needed. Thanks!

**Comment by devansh-webkul** [comment]
Thanks! Just one last question is this tested on postgress also.

**Comment by SopsoN** [comment]
@devansh-webkul I’ve just pushed a commit with test updates to make them compatible with PostgreSQL. The core functionality worked without any issues. Please check if everything is okay now.

**Comment by devansh-webkul** [comment]
Thanks for putting this together. After reviewing, I think a few structural things need addressing before this can land: the product-type handling is hardcoded to simple/configurable and doesn't cover grouped, bundle, virtual, downloadable, or booking; the event listener and migration use legacy syntax and PostgreSQL compatibility isn't fully handled — the FK type mismatch in the migration will fail on PG CI. I'll take this over and push the updates on top of your branch so we can get it across the line.
